pkg/trace/agent/obfuscate.go

```diff
@@ -176,3 +177,46 @@ func (cco *ccObfuscator) MetaHook(k, v string) (newval string) {
 	}
 	return v
 }
+
+// MetaStructHook checks the tag with the given key and val and returns the final
+// value to be assigned to this tag.
+func (cco *ccObfuscator) MetaStructHook(k string, v []byte) (newval []byte) {
+	if k == "appsec" {
+		var obfuscated bool
+		var appsec pb.AppSecStruct
+		_, err := appsec.UnmarshalMsg(v)
```

Comment:
I wonder how will we manage multiple versions of the appsec event payloads in the future 🤔 ?

Reply:
The AppSec struct now uses protobuf and should support adding or removing fields.

Reply:
Ok all good because the msgpack wire-format can avoid unused fields 👍